config/config.go

```diff
@@ -16,29 +16,52 @@
 
 package config
 
-import "os"
+import (
+	"os"
+	"sync"
+	"time"
+)
 
 type Config struct {
 	RPCURL string
+
+	// Timeouts
+	DialTimeout      time.Duration
+	SubscribeTimeout time.Duration
 }
 
-// NewDefaultConfig returns a new config with default values. Default values can be overwritten with env variables,
-// most importantly RPC_URL for a custom endpoint.
-// TODO: rewrite as init function
-func NewDefaultConfig() Config {
-	c := Config{}
-	c.ExtractDefaultRPCURL()
-	return c
+var config *Config = nil
```

Comment:
We must avoid global variables. It's a code smell. I don't see why we need this.

Reply:
You are right, it's really not needed here. I removed the state, please review again.